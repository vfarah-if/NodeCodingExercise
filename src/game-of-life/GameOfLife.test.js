const GameOfLife = require('./GameOfLife');

describe('GameOfLife', () => {
	test('should generate an empty two by two board', () => {
		const gameOfLife = new GameOfLife(2);

		expect(gameOfLife.toString()).toMatchSnapshot();
	});

	test('should generate a two by two board with seeded data in all rows and columns', () => {
		const gameOfLife = new GameOfLife(2, [
			{ x: 0, y: 0 },
			{ x: 1, y: 0 },
			{ x: 0, y: 1 },
			{ x: 1, y: 1 },
		]);

		expect(gameOfLife.toString()).toMatchSnapshot();
	});

	test('should die from solitide with only one neighbour', () => {
		const gameOfLife = new GameOfLife(2, [
			{ x: 0, y: 0 },
			{ x: 1, y: 1 },
		]);

		expect(gameOfLife.toString()).toMatchSnapshot('InitialSeed');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');
	});

	test('should survive with two or three neighbours', () => {
		const gameOfLife = new GameOfLife(3, [
			{ x: 0, y: 0 },
			{ x: 1, y: 1 },
			{ x: 1, y: 2 },
		]);

		expect(gameOfLife.toString()).toMatchSnapshot('InitialSeed');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');
	});

	test('should generate a four by four block still life', () => {
		const gameOfLife = new GameOfLife(4, [
			{ x: 1, y: 1 },
			{ x: 2, y: 1 },
			{ x: 1, y: 2 },
			{ x: 2, y: 2 },
		]);

		expect(gameOfLife.toString()).toMatchSnapshot('InitialSeed');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');
	});

	test('should generate a three by three blinker oscilator', () => {
		const gameOfLife = new GameOfLife(3, [
			{ x: 0, y: 1 },
			{ x: 1, y: 1 },
			{ x: 2, y: 1 },
		]);

		expect(gameOfLife.toString()).toMatchSnapshot('InitialSeed');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');
	});

	test('should generate a six by six blinker oscilator', () => {
		const gameOfLife = new GameOfLife(6, [
			{ x: 1, y: 1 },
			{ x: 2, y: 1 },
			{ x: 1, y: 2 },
			{ x: 2, y: 2 },
			{ x: 3, y: 3 },
			{ x: 4, y: 3 },
			{ x: 3, y: 4 },
			{ x: 4, y: 4 },
		]);

		expect(gameOfLife.toString()).toMatchSnapshot('InitialSeed');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');

		gameOfLife.generate();

		expect(gameOfLife.toString()).toMatchSnapshot('Generate');
	});
});
