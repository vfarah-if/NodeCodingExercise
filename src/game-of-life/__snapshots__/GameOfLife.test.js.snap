// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOfLife should die from solitide with only one neighbour: Generate 1`] = `
" |   |   | 
 |   |   | "
`;

exports[`GameOfLife should die from solitide with only one neighbour: InitialSeed 1`] = `
" | X |   | 
 |   | X | "
`;

exports[`GameOfLife should generate a four by four block still life: Generate 1`] = `
" |   |   |   |   | 
 |   | X | X |   | 
 |   | X | X |   | 
 |   |   |   |   | "
`;

exports[`GameOfLife should generate a four by four block still life: InitialSeed 1`] = `
" |   |   |   |   | 
 |   | X | X |   | 
 |   | X | X |   | 
 |   |   |   |   | "
`;

exports[`GameOfLife should generate a six by six blinker oscilator: Generate 1`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X |   |   |   |   | 
 |   |   |   |   | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`GameOfLife should generate a six by six blinker oscilator: Generate 2`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X | X |   |   |   | 
 |   |   |   | X | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`GameOfLife should generate a six by six blinker oscilator: Generate 3`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X |   |   |   |   | 
 |   |   |   |   | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`GameOfLife should generate a six by six blinker oscilator: InitialSeed 1`] = `
" |   |   |   |   |   |   | 
 |   | X | X |   |   |   | 
 |   | X | X |   |   |   | 
 |   |   |   | X | X |   | 
 |   |   |   | X | X |   | 
 |   |   |   |   |   |   | "
`;

exports[`GameOfLife should generate a three by three blinker oscilator: Generate 1`] = `
" |   | X |   | 
 |   | X |   | 
 |   | X |   | "
`;

exports[`GameOfLife should generate a three by three blinker oscilator: Generate 2`] = `
" |   |   |   | 
 | X | X | X | 
 |   |   |   | "
`;

exports[`GameOfLife should generate a three by three blinker oscilator: Generate 3`] = `
" |   | X |   | 
 |   | X |   | 
 |   | X |   | "
`;

exports[`GameOfLife should generate a three by three blinker oscilator: InitialSeed 1`] = `
" |   |   |   | 
 | X | X | X | 
 |   |   |   | "
`;

exports[`GameOfLife should generate a two by two board with seeded data in all rows and columns 1`] = `
" | X | X | 
 | X | X | "
`;

exports[`GameOfLife should generate an empty two by two board 1`] = `
" |   |   | 
 |   |   | "
`;

exports[`GameOfLife should survive with two or three neighbours: Generate 1`] = `
" |   |   |   | 
 | X | X |   | 
 |   |   |   | "
`;

exports[`GameOfLife should survive with two or three neighbours: InitialSeed 1`] = `
" | X |   |   | 
 |   | X |   | 
 |   | X |   | "
`;
